phpBB/phpbb/db/doctrine/postgresql_platform.php

```diff
@@ -0,0 +1,179 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\db\doctrine;
+
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Platforms\PostgreSQL94Platform;
+use Doctrine\DBAL\Schema\Index;
+use Doctrine\DBAL\Schema\Sequence;
+use Doctrine\DBAL\Schema\Table;
+use Doctrine\DBAL\Types\BigIntType;
+use Doctrine\DBAL\Types\IntegerType;
+use Doctrine\DBAL\Types\SmallIntType;
+use Doctrine\DBAL\Types\Type;
+
+/**
+ * PostgreSQL specific schema restrictions for BC.
+ *
+ * Doctrine is using SERIAL which auto creates the sequences with
+ * a name different from the one our driver is using. So in order
+ * to stay compatible with the existing DB we have to change its
+ * naming and not ours.
+ */
+class postgresql_platform extends PostgreSQL94Platform
+{
+	/**
+	 * {@inheritdoc}
+	 */
+	public function getIdentitySequenceName($tableName, $columnName)
```

Comment:
I would also opt for adding the return type, similar to the oracle platform.

Reply:
Yes, subclass can define anything that's valid. Adding a return type hint is allowed, not having one when the interface / parent class defines it is not.